search-service/src/main/kotlin/com/egm/stellio/search/entity/service/EntityQueryService.kt

```diff
@@ -251,15 +267,34 @@ class EntityQueryService(
             .bind("entity_id", entityId)
             .oneToResult { it["entityExists"] as Boolean }
             .flatMap {
-                if (it && !inverse || !it && inverse)
+                if (it)
                     Unit.right()
-                else if (it)
-                    AlreadyExistsException(entityAlreadyExistsMessage(entityId.toString())).left()
                 else
                     ResourceNotFoundException(entityNotFoundMessage(entityId.toString())).left()
             }
     }
 
+    /**
+     * Used for checks before creating a (temporal) entity. Allows to know if the entity does not exist,
+     * or, if it exists, whether it is currently deleted (in which case, it may be possible to create it again
+     * if authorized)
+     */
+    suspend fun getEntityState(
+        entityId: URI
+    ): Either<APIException, Pair<URI, ZonedDateTime?>> {
```

Comment:
Why are we returning the id? Is it different from the entityId argument? Could also use a better name maybe something like getEntityDeletionDate .

Reply:
I don't think `getEntityDletionDate()` is fit because if the entity doesn't exist it returns a `ResourceNotFoundException` -> therefore allowing to know the state of the entity
Maybe explain more in the comment instead?

Reply:
In the other hand the function is returning the deletedAt property so it is getting the deletionDate. I see that it is only used to know the actual deleted state of the entity so it could return a boolean (or enum) that represent if it was deleted. (in this case a possible name will be isMarkedDeleted)

Reply:
yep, I will change that. and yes, returning the id of the entity is useless.